fish-rust/src/common.rs

```diff
@@ -1376,7 +1375,7 @@ pub fn narrow_string_safe(buff: &mut [u8; 64], s: &wstr) {
 }
 
 /// Stored in blocks to reference the file which created the block.
-pub type FilenameRef = Rc<WString>;
+pub type FilenameRef = Arc<WString>;
```

Comment:
Is the atomicity needed everywhere `FilenameRef` was already being used? (i.e. are there places where `Rc` is good enough but new places where `Arc` is required or does everything actually need `Arc` but the code exposing that need wasn't written until now?)

If so, getting rid of the alias and just using `Arc` or `Rc` as needed would be better than paying the `Arc` price everywhere.

Reply:
The reason to use Arc here is that, under concurrent execution, we may have two threads with the same filename reference, for example piping one block into another.

Reply:
I understand; I was asking if there are any *other* usage sites  that are exclusively intra-thread rather than cross-thread, though. I’ll grep the code base later when I have a chance.